src/network/networkd-route.c

```diff
@@ -87,20 +87,68 @@ static const char * const route_table_table[] = {
         [RT_TABLE_LOCAL]   = "local",
 };
 
-DEFINE_STRING_TABLE_LOOKUP(route_table, int);
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP(route_table, int);
 
-#define ROUTE_TABLE_STR_MAX CONST_MAX(DECIMAL_STR_MAX(int), STRLEN("default") + 1)
-static const char *format_route_table(int table, char *buf, size_t size) {
+int route_table_from_string_full(const Manager *m, const char *s, uint32_t *ret) {
+        uint32_t t;
+        int r;
+
+        assert(m);
+        assert(s);
+        assert(ret);
+
+        r = route_table_from_string(s);
+        if (r >= 0) {
+                *ret = (uint32_t) r;
+                return 0;
+        }
+
+        t = PTR_TO_UINT32(hashmap_get(m->route_table_numbers_by_name, s));
+        if (t != 0) {
+                *ret = t;
+                return 0;
+        }
+
+        r = safe_atou32(s, &t);
+        if (r < 0)
+                return r;
+
+        if (t == 0)
+                return -ERANGE;
+
+        *ret = t;
+        return 0;
+}
+
+int route_table_to_string_full(const Manager *m, uint32_t table, char **ret) {
+        _cleanup_free_ char *str = NULL;
         const char *s;
-        char *p = buf;
+
+        assert(m);
+        assert(ret);
+
+        if (table == 0)
+                return -EINVAL;
```

Comment:
assert_return() maybe?

Reply:
or is it expected to get 0 here from time to time?

Reply:
> or is it expected to get 0 here from time to time?

Should not, but might be. Currently, this is only used in debug logs. So, it should not trigger any assertions about `table`.

Note, usually `xxx_to_string()` does not trigger any assertion when an input is invalid, but simply returns NULL. Also similar for `xxx_to_string_alloc()`. So, I think it is better to simply return negative errno here.

Reply:
Ok - note that I mentioned assert_return, ie: don't crash crash but logs an unexpected error

Reply:
ok, sounds good